### DISCUSSION TITLE
want to know the capabilities of moose framework

### DISCUSSION THREAD

**Discussion by habibhrk** [arfc/moltres#247] [Q&A Getting started] [answered]
I want to know whether the current version of moose/moltres supports moving mesh or lagrangian type of formulations to simulate thermal expansion or like compression.

**Comment by smpark7** [comment]
Hi @habibhrk, MOOSE has capabilities for modeling deformation and other phenomena under their Tensor Mechanics module. I personally don't have experience with modeling those physics but I think @js-jixu has had some success with coupling deformation with neutron diffusion modeling from Moltres

**Reply by js-jixu** [reply, marked as answer]
Hi, @habibhrk . As @smpark7  said, I have previously tried to couple the neutron field, displacement field, and thermal hydraulic field together in a fully coupled method. But saddly it cannot solve. Specific information can be found in this [discussion](https://github.com/idaholab/moose/discussions/22721)

**Reply by habibhrk** [reply]
Thanks @smpark7  and @js-jixu  for taking out time to address my query!
Sir, Actually i want to know whether modelling high energy system for example Z-Pinch phenomenon is possible while using moose framework or not ??

**Reply by smpark7** [reply]
@js-jixu I'm sorry to hear that you couldn't get it to work for your use case.

@habibhrk That scenario is well beyond the scope for Moltres. I recommend asking on the moose github repository or maybe [Zapdos](https://github.com/shannon-lab/zapdos) which specializes in plasma simulations.

**Reply by habibhrk** [reply]
Thanks @smpark7  I will look into it, Zapdos.

**Comment by smpark7** [comment]
For future users encountering this discussion, I want to add that modeling geometric deformation is complicated because we'd need to incorporate the effects of the resultant density changes in the neutron cross section. At this time, Moltres only accounts for isotropic density changes by interpolating cross section values generated at various temperature-dependent densities.

Even if we could couple Moltres to Tensor Mechanics through some other coupling method (e.g., MultiApp coupling), we'd be neglecting some of the density-induced reactivity feedback. However, if anyone wishes to work on implementing this feature, feel free to reach out to us.

**Reply by js-jixu** [reply]
Correct reminder! And I think we must pay attention to the `power` issue after the volume change of the element.
